### PR TITLE
Bug 1205588 - [Settings] Define MediaStorage using new syntax provided by Observable, r=yzen

### PR DESCRIPTION
**Pull request by gasolin**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d0dcf56ee1de2b48d19f17bab93168634003c706

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=218afb584d243c431bd9af305179e19f9b1b7b2f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=33af6c1addf41c792776b42d4a767064773da248

File: apps/settings/js/panels/root/storage_usb_item.js [hunk 1]
```diff
@@ -23,13 +22,11 @@ define(function(require) {
     this._enabled = false;
     this._elements = elements;
     this._umsSettingKey = 'ums.enabled';
-    // XXX media related attributes
-    this._defaultMediaVolume = null;
-    this._defaultVolumeState = 'available';
```
Comment (gasolin) [inline apps/settings/js/panels/root/storage_usb_item.js]: is replaced by observable `volumeState`

File: apps/settings/js/panels/root/storage_usb_item.js [hunk 2]
```diff
@@ -176,49 +166,12 @@ define(function(require) {
...
-    },
-
-    _updateVolumeState: function storage_updateVolumeState(volume, state) {
-      this._defaultVolumeState = state;
+    _updateVolumeState: function storage_updateVolumeState() {
       this._updateUmsDesc();
-      switch (state) {
+      switch (MediaStorage.volumeState) {
         case 'available':
-          this._updateMediaFreeSpace(volume);
```
Comment (gasolin) [inline apps/settings/js/panels/root/storage_usb_item.js]: don't need to query freespace here, so we can simplify the call

File: apps/settings/js/panels/root/storage_usb_item.js [hunk 3]
```diff
@@ -236,37 +189,12 @@ define(function(require) {
...
     _lockMediaStorageMenu: function storage_setMediaMenuState(lock) {
       if (lock) {
         this._elements.mediaStorageSection.setAttribute('aria-disabled', true);
       } else {
         this._elements.mediaStorageSection.removeAttribute('aria-disabled');
       }
-    },
-
-    // util function
-    _getDefaultVolume: function storage_getDefaultVolume(name) {
```
Comment (gasolin) [inline apps/settings/js/panels/root/storage_usb_item.js]: moved to `MediaStorage.js`

File: apps/settings/js/panels/root/storage_usb_item.js [hunk 4]
```diff
@@ -176,49 +166,12 @@ define(function(require) {
...
-      }
-      this._defaultMediaVolume = this._getDefaultVolume(defaultName);
-      if (this._defaultMediaVolume) {
-        this._defaultMediaVolume.addEventListener('change', this);
-      }
-      this._updateMediaStorageInfo();
-    },
-
-    // Media Storage
-    _updateMediaStorageInfo: function storage_updateMediaStorageInfo() {
```
Comment (gasolin) [inline apps/settings/js/panels/root/storage_usb_item.js]: moved to `MediaStorage.js`

File: apps/settings/js/panels/root/storage_usb_item.js [hunk 5]
```diff
@@ -176,49 +166,12 @@ define(function(require) {
         }
     },
 
-    // XXX media related functions
-    _mediaVolumeChangeHandler:
```
Comment (gasolin) [inline apps/settings/js/panels/root/storage_usb_item.js]: moved to `MediaStorage.js`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3be72d85b2fa624de21e6812633313a45d49e232

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c90948edaaea9e89fe47a3c32466bee99c1047cc

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a15f10896590310dd13036ba538f1eb4b0258a0e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f23f0ffcd82656079ef35b11307e34007bcde540

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f92482ea6750c6f96ae6fb5125e1ec2867d8295c

File: apps/settings/js/panels/root/storage_usb_item.js [hunk 6]
```diff
@@ -1,13 +1,21 @@
...
+ * Will disable Media Storage panel when UMS enabled.
  */
 define(function(require) {
   'use strict';
 
   var SettingsListener = require('shared/settings_listener');
+  var MediaStorage = require('modules/media_storage');
+
+  var _debug = false;
+  var Debug = function() {};
```
Comment (yzen) [inline apps/settings/js/panels/root/storage_usb_item.js]: can we have a lower case name; it will be misread as a constructor.

Comment (gasolin) [inline apps/settings/js/panels/root/storage_usb_item.js]: filed bug 1206984 to replace all these kind of syntax

**Pull request merged**